mapie/calibration.py

```diff
@@ -0,0 +1,559 @@
+        specifically fitted calibrator. Note that if there is no calibrator
+        for a the specific class, then we simply output the not calibrated
+        values.
+
+        Note that if the calibrated score prediction is 0, there would be an
+        issue when finding the class it belongs to. We set it equal to
+        {np.finfo(np.float64).eps} as this would likely not have large impact
+        on the calibration scores, yet, would set the maximum to the correct
+        label.
+
+
+        Parameters
+        ----------
+        item : int
+            The defined subset to be calibrated.
+        calibrated_values : NDArray
+            Array of calibrated values.
+        max_prob : NDArray of shape (n_samples,)
+            Values to be calibrated.
+        y_pred : NDArray of shape (n_samples,)
+            Indicator of the values to be calibrated.
+        calibrators : Dict[RegressorMixin] of length n_classes
+            Dictionary of all the previously fitted calibrators.
+
+        Returns
+        -------
+        NDArray
+            Updated calibrated values from the predictions of the calibrators.
+
+        Raises
+        ------
+        Warnings
+            If there has not been a fitted calibrator for the specific class.
+        """
+        correct_label = np.where(y_pred.ravel() == item)[0].ravel()
+        if item not in calibrators:
+            calibrated_values[
+                correct_label, idx
+                ] = max_prob[correct_label].ravel()
+            warnings.warn(
+                "WARNING: This predicted label {item} has not been seen during"
+                + " the calibration and therefore scores will remain"
+                + " unchanged."
+            )
+        else:
+            EPSILON = np.finfo(np.float64).eps
+            calibrator_ = calibrators[item]
+            preds_ = calibrator_.predict(max_prob[correct_label])
+            idx_zero_pred = np.where(preds_ < EPSILON)[0]
+            preds_[idx_zero_pred] = EPSILON
+            calibrated_values[correct_label, idx] = preds_
+        return calibrated_values
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[NDArray] = None,
+        calib_size: Optional[float] = 0.33,
+        random_state: Optional[Union[int, np.random.RandomState, None]] = None,
+        shuffle: Optional[bool] = True,
+        stratify: Optional[ArrayLike] = None,
+    ) -> MapieCalibrator:
+        """
+        Fit estimator will calibrate the predicted probabilities from the
+        output of a classifier.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Training data.
+        y : ArrayLike of shape (n_samples,)
+            Training labels.
+        sample_weight : Optional[ArrayLike] of shape (n_samples,)
+            Sample weights for fitting the out-of-fold models.
```

Comment:
Which out-of-fold? I fear it might come from MapieRegressor?

Reply:
Again, this definition of sample weight is constant across MAPIE. I believe we could discuss changing it, but then this refacto should be done across all classes.